src/posix.mak

```diff
@@ -131,7 +131,7 @@ endif
 
 DEPS = $(patsubst %.o,%.deps,$(DMD_OBJS))
 
-RUN_BUILD = $(GENERATED)/build HOST_DMD="$(HOST_DMD)" CXX="$(HOST_CXX)" OS=$(OS) BUILD=$(BUILD) MODEL=$(MODEL) AUTO_BOOTSTRAP="$(AUTO_BOOTSTRAP)" DOCDIR="$(DOCDIR)" STDDOC="$(STDDOC)" DOC_OUTPUT_DIR="$(DOC_OUTPUT_DIR)" MAKE="$(MAKE)" --called-from-make
+RUN_BUILD = $(GENERATED)/build HOST_DMD="$(HOST_DMD)" CXX="$(HOST_CXX)" OS=$(OS) BUILD=$(BUILD) MODEL=$(MODEL) AUTO_BOOTSTRAP="$(AUTO_BOOTSTRAP)" DOCDIR="$(DOCDIR)" STDDOC="$(STDDOC)" DOC_OUTPUT_DIR="$(DOC_OUTPUT_DIR)" MAKE="$(MAKE)" ENABLE_RELEASE=$(ENABLE_RELEASE) ENABLE_DEBUG=$(ENABLE_DEBUG) --called-from-make
```

Comment:
Its not required to replicate these flags here, `build.d` fetches them from the environment

Reply:
they fetch them if they are environment variables, but not if they are Make variables.

Reply:
`ENABLE_RELASE` is not definied within `posix.mak`. `ENABLE_DEBUG` on the other hand only once for `BUILD=DEBUG` albeit it hould be handled in `build.d` (this can be fixed in another PR)

Reply:
if you run `make -f posix.mak ENABLE_RELEASE=...` then it will not get picked up by `build.d` because in that case it's not an environment variable.

Reply:
Using `GNU Make 4.1`:

```
make -f posix.mak ENABLE_LTO=1
make -C src -f posix.mak all
make[1]: Entering directory '/mnt/e/repos/Dlang/forks/dmd/src'
no cpu specified, assuming X86
../generated/build HOST_DMD="dmd" CXX="c++" OS=linux BUILD=release MODEL=64 AUTO_BOOTSTRAP="" DOCDIR="" STDDOC="" DOC_OUTPUT_DIR="" MAKE="make" --called-from-make -v --dry-run ../generated/linux/release/64/dmd
Run: /usr/bin/dmd --version
================================================================================
... [Other variables]...
ENABLE_LTO=1
================================================================================
....
```

`ENABLE_LTO` is propagated to `build.d`. Are we using different `make` implementations?

Reply:
```
[marler8997@nixos:~/ddev1/dmd]$ make --version
GNU Make 4.2.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

I added `-v` to the `RUN_BUILD` command and setting `ENABLE_LTO=1` does not propogate to `build.d` for me:
```
[marler8997@nixos:~/ddev1/dmd]$ make -f posix.mak ENABLE_LTO=1
make -C src -f posix.mak all
make[1]: Entering directory '/home/marler8997/ddev1/dmd/src'
no cpu specified, assuming X86
dmd -of../generated/build build.d
../generated/build HOST_DMD="dmd" CXX="c++" OS=linux BUILD=release MODEL=64 AUTO_BOOTSTRAP="" DOCDIR="" STDDOC="" DOC_OUTPUT_DIR="" MAKE="make" ENABLE_RELEASE= ENABLE_DEBUG= --called-from-make -v ../generated/linux/release/64/dmd
Run: /run/current-system/sw/bin/dmd --version
================================================================================
STDDOC=
TOLF=tolf
AUTO_BOOTSTRAP=
MODEL=64
ROOT=/home/marler8997/ddev1/dmd/src/dmd/root
DMD_PATH=/home/marler8997/ddev1/dmd/generated/linux/release/64/dmd
DETAB=detab
SYSCONFDIR=/etc
GENERATED=/home/marler8997/ddev1/dmd/generated
EX=/home/marler8997/ddev1/dmd/src/examples
RES=/home/marler8997/ddev1/dmd/res
MODEL_FLAG=-m64
CXX=c++
TARGET_CPU=X86
BUILD=release
HOST_DMD_KIND=dmd
MAKE=make
GIT_HOME=https://github.com/dlang
OS=linux
ZIP=zip
CXX_KIND=g++
D=/home/marler8997/ddev1/dmd/src/dmd
PIC_FLAG=-fPIC
G=/home/marler8997/ddev1/dmd/generated/linux/release/64
C=/home/marler8997/ddev1/dmd/src/dmd/backend
DOCDIR=/home/marler8997/ddev1/dmd/src
INSTALL=/home/marler8997/ddev1/dmd/install
DOC_OUTPUT_DIR=
ENABLE_DEBUG=
HOST_DMD_RUN=/run/current-system/sw/bin/dmd
TOOLS_DIR=/home/marler8997/ddev1/tools
HOST_DMD_PATH=/run/current-system/sw/bin/dmd
ENABLE_RELEASE=
GIT=git
DOCSRC=/home/marler8997/ddev1/dmd/dlang.org
HOST_DMD=dmd
TMP=/tmp
================================================================================
Run: git describe --dirty
(DC) BACKEND_OBJ
Skipping build of /home/marler8997/ddev1/dmd/generated/linux/release/64/SYSCONFDIR.imp as its condition returned false
Run: /run/current-system/sw/bin/dmd -c -of/home/marler8997/ddev1/dmd/generated/linux/release/64/backend.o -betterC -version=MARS -w -de -fPIC -m64 -J/home/marler8997/ddev1/dmd/generated/linux/release/64 -dip25 -g /home/marler8997/ddev1/dmd/src/dmd/backend/backend.d /home/marler8997/ddev1/dmd/src/dmd/backend/bcomplex.d /home/marler8997/ddev1/dmd/src/dmd/backend/evalu8.d /home/marler8997/ddev1/dmd/src/dmd/backend/divcoeff.d /home/marler8997/ddev1/dmd/src/dmd/backend/dvec.d /home/marler8997/ddev1/dmd/src/dmd/backend/go.d /home/marler8997/ddev1/dmd/src/dmd/backend/gsroa.d /home/marler8997/ddev1/dmd/src/dmd/backend/glocal.d /home/marler8997/ddev1/dmd/src/dmd/backend/gdag.d /home/marler8997/ddev1/dmd/src/dmd/backend/gother.d /home/marler8997/ddev1/dmd/src/dmd/backend/gflow.d /home/marler8997/ddev1/dmd/src/dmd/backend/out.d /home/marler8997/ddev1/dmd/src/dmd/backend/gloop.d /home/marler8997/ddev1/dmd/src/dmd/backend/compress.d /home/marler8997/ddev1/dmd/src/dmd/backend/cgelem.d /home/marler8997/ddev1/dmd/src/dmd/backend/cgcs.d /home/marler8997/ddev1/dmd/src/dmd/backend/ee.d /home/marler8997/ddev1/dmd/src/dmd/backend/cod4.d /home/marler8997/ddev1/dmd/src/dmd/backend/cod5.d /home/marler8997/ddev1/dmd/src/dmd/backend/nteh.d /home/marler8997/ddev1/dmd/src/dmd/backend/blockopt.d /home/marler8997/ddev1/dmd/src/dmd/backend/mem.d /home/marler8997/ddev1/dmd/src/dmd/backend/cg.d /home/marler8997/ddev1/dmd/src/dmd/backend/cgreg.d /home/marler8997/ddev1/dmd/src/dmd/backend/dtype.d /home/marler8997/ddev1/dmd/src/dmd/backend/debugprint.d /home/marler8997/ddev1/dmd/src/dmd/backend/fp.d /home/marler8997/ddev1/dmd/src/dmd/backend/symbol.d /home/marler8997/ddev1/dmd/src/dmd/backend/elem.d /home/marler8997/ddev1/dmd/src/dmd/backend/dcode.d /home/marler8997/ddev1/dmd/src/dmd/backend/cgsched.d /home/marler8997/ddev1/dmd/src/dmd/backend/cg87.d /home/marler8997/ddev1/dmd/src/dmd/backend/cgxmm.d /home/marler8997/ddev1/dmd/src/dmd/backend/cgcod.d /home/marler8997/ddev1/dmd/src/dmd/backend/cod1.d /home/marler8997/ddev1/dmd/src/dmd/backend/cod2.d /home/marler8997/ddev1/dmd/src/dmd/backend/cod3.d /home/marler8997/ddev1/dmd/src/dmd/backend/cv8.d /home/marler8997/ddev1/dmd/src/dmd/backend/dcgcv.d /home/marler8997/ddev1/dmd/src/dmd/backend/pdata.d /home/marler8997/ddev1/dmd/src/dmd/backend/util2.d /home/marler8997/ddev1/dmd/src/dmd/backend/var.d /home/marler8997/ddev1/dmd/src/dmd/backend/md5.d /home/marler8997/ddev1/dmd/src/dmd/backend/backconfig.d /home/marler8997/ddev1/dmd/src/dmd/backend/ph2.d /home/marler8997/ddev1/dmd/src/dmd/backend/drtlsym.d /home/marler8997/ddev1/dmd/src/dmd/backend/dwarfeh.d /home/marler8997/ddev1/dmd/src/dmd/backend/ptrntab.d /home/marler8997/ddev1/dmd/src/dmd/backend/dvarstats.d /home/marler8997/ddev1/dmd/src/dmd/backend/dwarfdbginf.d /home/marler8997/ddev1/dmd/src/dmd/backend/cgen.d /home/marler8997/ddev1/dmd/src/dmd/backend/os.d /home/marler8997/ddev1/dmd/src/dmd/backend/goh.d /home/marler8997/ddev1/dmd/src/dmd/backend/barray.d /home/marler8997/ddev1/dmd/src/dmd/backend/cgcse.d /home/marler8997/ddev1/dmd/src/dmd/backend/elpicpie.d /home/marler8997/ddev1/dmd/src/dmd/backend/machobj.d /home/marler8997/ddev1/dmd/src/dmd/backend/elfobj.d /home/marler8997/ddev1/dmd/src/dmd/backend/aarray.d
(TX) VERSION
(DC) LEXER_OBJ
Run: /run/current-system/sw/bin/dmd -c -of/home/marler8997/ddev1/dmd/generated/linux/release/64/lexer.o -vtls -version=MARS -w -de -fPIC -m64 -J/home/marler8997/ddev1/dmd/generated/linux/release/64 -dip25 -g dmd/console.d dmd/entity.d dmd/errors.d dmd/filecache.d dmd/globals.d dmd/id.d dmd/identifier.d dmd/lexer.d dmd/tokens.d dmd/utf.d dmd/root/array.d dmd/root/bitarray.d dmd/root/ctfloat.d dmd/root/file.d dmd/root/filename.d dmd/root/hash.d dmd/root/outbuffer.d dmd/root/port.d dmd/root/region.d dmd/root/rmem.d dmd/root/rootobject.d dmd/root/stringtable.d
(DC) DMD
Run: /run/current-system/sw/bin/dmd -of/home/marler8997/ddev1/dmd/generated/linux/release/64/dmd -vtls -J/home/marler8997/ddev1/dmd/res -version=MARS -w -de -fPIC -m64 -J/home/marler8997/ddev1/dmd/generated/linux/release/64 -dip25 -g dmd/access.d dmd/aggregate.d dmd/aliasthis.d dmd/apply.d dmd/argtypes.d dmd/argtypes_sysv_x64.d dmd/arrayop.d dmd/arraytypes.d dmd/ast_node.d dmd/astbase.d dmd/astcodegen.d dmd/attrib.d dmd/blockexit.d dmd/builtin.d dmd/canthrow.d dmd/cli.d dmd/clone.d dmd/compiler.d dmd/complex.d dmd/cond.d dmd/constfold.d dmd/cppmangle.d dmd/cppmanglewin.d dmd/ctfeexpr.d dmd/ctorflow.d dmd/dcast.d dmd/dclass.d dmd/declaration.d dmd/delegatize.d dmd/denum.d dmd/dimport.d dmd/dinifile.d dmd/dinterpret.d dmd/dmacro.d dmd/dmangle.d dmd/dmodule.d dmd/doc.d dmd/dscope.d dmd/dstruct.d dmd/dsymbol.d dmd/dsymbolsem.d dmd/dtemplate.d dmd/dversion.d dmd/env.d dmd/escape.d dmd/expression.d dmd/expressionsem.d dmd/func.d dmd/hdrgen.d dmd/impcnvtab.d dmd/imphint.d dmd/init.d dmd/initsem.d dmd/inline.d dmd/inlinecost.d dmd/intrange.d dmd/json.d dmd/lambdacomp.d dmd/lib.d dmd/libelf.d dmd/libmach.d dmd/libmscoff.d dmd/libomf.d dmd/link.d dmd/mars.d dmd/mtype.d dmd/nogc.d dmd/nspace.d dmd/objc.d dmd/opover.d dmd/optimize.d dmd/parse.d dmd/parsetimevisitor.d dmd/permissivevisitor.d dmd/printast.d dmd/safe.d dmd/sapply.d dmd/scanelf.d dmd/scanmach.d dmd/scanmscoff.d dmd/scanomf.d dmd/semantic2.d dmd/semantic3.d dmd/sideeffect.d dmd/statement.d dmd/statement_rewrite_walker.d dmd/statementsem.d dmd/staticassert.d dmd/staticcond.d dmd/strictvisitor.d dmd/target.d dmd/templateparamsem.d dmd/traits.d dmd/transitivevisitor.d dmd/typesem.d dmd/typinf.d dmd/utils.d dmd/visitor.d dmd/foreachvar.d dmd/irstate.d dmd/toctype.d dmd/glue.d dmd/gluelayer.d dmd/todt.d dmd/tocsym.d dmd/toir.d dmd/dmsc.d dmd/tocvdebug.d dmd/s2ir.d dmd/toobj.d dmd/e2ir.d dmd/eh.d dmd/iasm.d dmd/iasmdmd.d dmd/iasmgcc.d dmd/objc_glue.d dmd/backend/cc.d dmd/backend/cdef.d dmd/backend/cgcv.d dmd/backend/code.d dmd/backend/cv4.d dmd/backend/dt.d dmd/backend/el.d dmd/backend/global.d dmd/backend/obj.d dmd/backend/oper.d dmd/backend/outbuf.d dmd/backend/rtlsym.d dmd/backend/code_x86.d dmd/backend/iasm.d dmd/backend/codebuilder.d dmd/backend/ty.d dmd/backend/type.d dmd/backend/exh.d dmd/backend/mach.d dmd/backend/mscoff.d dmd/backend/dwarf.d dmd/backend/dwarf2.d dmd/backend/xmm.d dmd/backend/dlist.d dmd/backend/melf.d dmd/backend/varstats.di dmd/root/aav.d dmd/root/longdouble.d dmd/root/man.d dmd/root/response.d dmd/root/speller.d dmd/root/string.d dmd/root/strtold.d ../generated/linux/release/64/lexer.o ../generated/linux/release/64/backend.o
Success
../generated/build HOST_DMD="dmd" CXX="c++" OS=linux BUILD=release MODEL=64 AUTO_BOOTSTRAP="" DOCDIR="" STDDOC="" DOC_OUTPUT_DIR="" MAKE="make" ENABLE_RELEASE= ENABLE_DEBUG= --called-from-make -v ../generated/linux/release/64/dmd.conf
Run: /run/current-system/sw/bin/dmd --version
================================================================================
STDDOC=
TOLF=tolf
AUTO_BOOTSTRAP=
MODEL=64
ROOT=/home/marler8997/ddev1/dmd/src/dmd/root
DMD_PATH=/home/marler8997/ddev1/dmd/generated/linux/release/64/dmd
DETAB=detab
SYSCONFDIR=/etc
GENERATED=/home/marler8997/ddev1/dmd/generated
EX=/home/marler8997/ddev1/dmd/src/examples
RES=/home/marler8997/ddev1/dmd/res
MODEL_FLAG=-m64
CXX=c++
TARGET_CPU=X86
BUILD=release
HOST_DMD_KIND=dmd
MAKE=make
GIT_HOME=https://github.com/dlang
OS=linux
ZIP=zip
CXX_KIND=g++
D=/home/marler8997/ddev1/dmd/src/dmd
PIC_FLAG=-fPIC
G=/home/marler8997/ddev1/dmd/generated/linux/release/64
C=/home/marler8997/ddev1/dmd/src/dmd/backend
DOCDIR=/home/marler8997/ddev1/dmd/src
INSTALL=/home/marler8997/ddev1/dmd/install
DOC_OUTPUT_DIR=
ENABLE_DEBUG=
HOST_DMD_RUN=/run/current-system/sw/bin/dmd
TOOLS_DIR=/home/marler8997/ddev1/tools
HOST_DMD_PATH=/run/current-system/sw/bin/dmd
ENABLE_RELEASE=
GIT=git
DOCSRC=/home/marler8997/ddev1/dmd/dlang.org
HOST_DMD=dmd
TMP=/tmp
================================================================================
(TX) DMD_CONF
Success
make[1]: Leaving directory '/home/marler8997/ddev1/dmd/src'
```

Reply:
Looks like Make actually does set variables set with `make VAR=VALUE` as environment variables.  So no issue here and I can remove these extra variables.